File: ExtraCredit/Thirdlightbulb.java
public class Thirdlightbulb {
 
    public static long fib(int n) {
        if (n <= 1) {
            return n;
        }
        long[] fibCache = new long[n + 1];
        fibCache[0] = 0;
        fibCache[1] = 1;
        
        for (int i = 2; i <= n; i++) {
            fibCache[i] = fibCache[i - 1] + fibCache[i - 2];
        }
        
        return fibCache[n];
    }

    public static void main(String[] args) {
        int n = 9; // Example: Finding the 9th Fibonacci number
        System.out.println("The " + n + "th Fibonacci number is: " + fib(n));
    } 
}

File: ExtraCredit/Thirdlightbulb.py
def fib(n):
    if n <= 1:
        return n
    fib_cache = [0, 1] + [0] * (n-1)
    
    for i in range(2, n + 1):
        fib_cache[i] = fib_cache[i - 1] + fib_cache[i - 2]
    
    return fib_cache[n]

# Example: Finding the 9th Fibonacci number
n = 9
print(f"The {n}th Fibonacci number is: {fib(n)}")
